pkg/streams/streamsGet.go

```diff
@@ -47,90 +65,133 @@ func getStreamsFromRestApi(
+	}
+
+	return formatterNames.String()
 }
 
 func getStreamByName(
 	streamName string,
 	apiClient *galasaapi.APIClient,
-	context *context.Context,
 	restApiVersion string,
 ) ([]galasaapi.Stream, error) {
 
 	var err error
 	var streamIn *galasaapi.Stream
 	var resp *http.Response
+	var context context.Context = nil
+	var streamsToReturn []galasaapi.Stream
 
-	apiCall := apiClient.StreamsAPIApi.GetStreamByName(*context, streamName).ClientApiVersion(restApiVersion)
+	apiCall := apiClient.StreamsAPIApi.GetStreamByName(context, streamName).ClientApiVersion(restApiVersion)
 
 	streamIn, resp, err = apiCall.Execute()
 
-	var statusCode int
-	if resp != nil {
-		defer resp.Body.Close()
-		statusCode = resp.StatusCode
-	}
+	if err == nil {
+		var statusCode int
+		if resp != nil {
+			defer resp.Body.Close()
+			statusCode = resp.StatusCode
+		}
 
-	if err != nil {
-		log.Println("getStreamsFromRestApi - Failed to retrieve list of test streams from API server")
-		err = galasaErrors.NewGalasaErrorWithHttpStatusCode(statusCode, galasaErrors.GALASA_ERROR_RETRIEVING_USER_LIST_FROM_API_SERVER, err.Error())
+		if statusCode != 200 {
+			log.Println("getStreamsFromRestApi - Failed to retrieve list of test streams from API server")
+			err = galasaErrors.NewGalasaErrorWithHttpStatusCode(statusCode, galasaErrors.GALASA_ERROR_RETRIEVING_STREAMS_FROM_API_SERVER, err.Error())
+		}
+		streamsToReturn = []galasaapi.Stream{*streamIn}
 	}
 
-	streamsToReturn := []galasaapi.Stream{*streamIn}
 	return streamsToReturn, err
 
 }
 
 func getAllStreams(
 	apiClient *galasaapi.APIClient,
-	context *context.Context,
 	restApiVersion string,
 ) ([]galasaapi.Stream, error) {
 
 	var err error
 	var streams []galasaapi.Stream
 	var resp *http.Response
+	var context context.Context = nil
 
-	apiCall := apiClient.StreamsAPIApi.GetStreams(*context).ClientApiVersion(restApiVersion)
+	apiCall := apiClient.StreamsAPIApi.GetStreams(context).ClientApiVersion(restApiVersion)
 
 	streams, resp, err = apiCall.Execute()
 
-	var statusCode int
-	if resp != nil {
-		defer resp.Body.Close()
-		statusCode = resp.StatusCode
-	}
+	if err == nil {
 
-	if err != nil {
-		log.Println("getStreamsFromRestApi - Failed to retrieve list of test streams from API server")
-		err = galasaErrors.NewGalasaErrorWithHttpStatusCode(statusCode, galasaErrors.GALASA_ERROR_RETRIEVING_STREAMS_FROM_API_SERVER, err.Error())
-	} else {
-		log.Printf("getUserDataFromRestApi - %v test streams collected", len(streams))
+		var statusCode int
+		if resp != nil {
+			defer resp.Body.Close()
+			statusCode = resp.StatusCode
+		}
 
+		if statusCode == 200 {
+			log.Printf("getUserDataFromRestApi - %v test streams collected", len(streams))
+		} else {
+			log.Println("getStreamsFromRestApi - Failed to retrieve list of test streams from API server")
+			err = galasaErrors.NewGalasaErrorWithHttpStatusCode(statusCode, galasaErrors.GALASA_ERROR_RETRIEVING_STREAMS_FROM_API_SERVER, err.Error())
+			statusCode = 500
+		}
```

Comment:
Same comment here re incorrect error messages. Setting statusCode to 500 won't be correct if the client failed to send the request.